File: Serenity Practice/HosbitalSerenity.Web/Modules/Hosbital/Speciality/SpecialityRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace HosbitalSerenity.Hosbital;

[ConnectionKey("Default"), Module("Hosbital"), TableName("Speciality")]
[DisplayName("Speciality"), InstanceName("Speciality")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript]
public sealed class SpecialityRow : Row<SpecialityRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Speciality Id"), Identity, IdProperty]
    public int? SpecialityId { get => fields.SpecialityId[this]; set => fields.SpecialityId[this] = value; }

    [DisplayName("Speciality Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string SpecialityName { get => fields.SpecialityName[this]; set => fields.SpecialityName[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field SpecialityId;
        public StringField SpecialityName;

    }
}

File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Hosbital/SpecialityRow.ts
﻿import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface SpecialityRow {
    SpecialityId?: number;
    SpecialityName?: string;
}

export abstract class SpecialityRow {
    static readonly idProperty = 'SpecialityId';
    static readonly nameProperty = 'SpecialityName';
    static readonly localTextPrefix = 'Hosbital.Speciality';
    static readonly lookupKey = 'Hosbital.Speciality';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<SpecialityRow>('Hosbital.Speciality') }
    static async getLookupAsync() { return getLookupAsync<SpecialityRow>('Hosbital.Speciality') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<SpecialityRow>();
}